src/librustdoc/html/highlight.rs

```diff
@@ -221,7 +276,7 @@ impl<'a> Classifier<'a> {
             } else if nb > 0 && has_ident {
                 return vec![(TokenKind::Ident, start, pos), (TokenKind::Colon, pos, pos + nb)];
             } else if nb > 0 {
-                return vec![(TokenKind::Colon, pos, pos + nb)];
+                return vec![(TokenKind::Colon, start, start + nb)];
```

Comment:
Why did this change?

Reply:
Same as above: we haven't found an ident yet so we want to return the whle "path". To make it more clear, I replaced `pos` with `start` (which are equal at this point).